src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/Metadata/IModelBindingMessages.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNet.Mvc.ModelBinding.Metadata
+{
+    /// <summary>
+    /// Messages for errors the model binding system detects.
+    /// </summary>
+    public interface IModelBindingMessages
```

Comment:
An interface for this kinda makes me sad. I understand why you ended up going that route